### PR TITLE
Accept and store de-identified hashes in Bulk Upload process

### LINKED ISSUE #1214: Accept deidentified PII records in bulk uploads
As a state agency developer, I want to share deidentified PII records in bulk uploads so that my state's participants can be matched to other state's participants.

Documentation portion (#1450) should be done prior to this task. 

AC:
- [ ] Update participant database and database schema
- [ ] Modify bulk upload to accept new CSV format
  - [ ] Require hash to be present
  - [ ] Validate hash is 128-char hexadecimal digest
- [ ] Remove validations for replaced fields
- [ ] Update test data - replace PII fields with valid hashes
- [ ] Update API documentation

### PR DESCRIPTION
**Pull request by echappen**
Closes #1214 

- [ ] run iac on `tts/dev` prior to merge

### Removes
- Removes `last`, `first`, `dob`, `ssn` columns from bulk upload csv

### Adds
- Add  `lds_hash` column to bulk upload csv

### Changes
- Makes `participant_id` column required on bulk upload csv

### CODE REVIEW

**Comment by ryanhofdotgov** [comment]
@echappen Can you say more about why IaC would need to be re-run?

File: etl/src/Piipan.Etl/BulkUpload.cs [hunk 1]
```diff
@@ -147,14 +147,10 @@ internal async static Task Load(IEnumerable<PiiRecord> records, DbProviderFactor
...
-                        cmd.CommandText = "INSERT INTO participants (last, first, middle, dob, ssn, upload_id, case_id, participant_id, benefits_end_date, recent_benefit_months, protect_location) " +
-                            "VALUES (@last, @first, @middle, @dob, @ssn, @upload_id, @case_id, @participant_id, @benefits_end_date, @recent_benefit_months::date[], @protect_location)";
-
-                        AddWithValue(cmd, DbType.String, "last", record.Last);
-                        AddWithValue(cmd, DbType.String, "first", (object)record.First ?? DBNull.Value);
-                        AddWithValue(cmd, DbType.String, "middle", (object)record.Middle ?? DBNull.Value);
-                        AddWithValue(cmd, DbType.DateTime, "dob", record.Dob);
-                        AddWithValue(cmd, DbType.String, "ssn", record.Ssn);
+                        cmd.CommandText = "INSERT INTO participants (lds_hash, upload_id, case_id, participant_id, benefits_end_date, recent_benefit_months, protect_location) " +
+                            "VALUES (@lds_hash, @ssn, @upload_id, @case_id, @participant_id, @benefits_end_date, @recent_benefit_months::date[], @protect_location)";
```
Comment (ryanhofdotgov) [inline etl/src/Piipan.Etl/BulkUpload.cs:151]: Need to delete @ssn

**Review by ryanhofdotgov — CHANGES_REQUESTED** [review]

**Comment by ryanhofdotgov** [comment]
Non-blocking, but it might be a good time to incorporate a basic integration test like @kingcomma did a while back for the match subsystem: [Piipan.Match.Orchestrator.IntegrationTests](https://github.com/18F/piipan/tree/dev/match/tests/Piipan.Match.Orchestrator.IntegrationTests)

**Comment by echappen** [comment]
> Can you say more about why IaC would need to be re-run?

@ryanhofdotgov Database would need to be dropped and recreated.

**Comment by ryanhofdotgov** [comment]
@echappen Ah – of course. Thanks! I have a little personal script that just re-creates the database bit, I need to make that into something formal.

**Review by ryanhofdotgov — APPROVED** [review]
Thanks! LGTM. 

I'll open some follow-on tasks:
- Bump bulk upload version number to v2
- Add basic database tests for ETL (would catch schema issues)

**Comment by echappen** [comment]
@ryanhofdotgov I opened a ticket for testing here: https://github.com/18F/piipan/issues/1487

**Pull request merged**